includes/apple-exporter/components/class-facebook.php

```diff
@@ -74,12 +97,18 @@ public static function node_matches( $node ) {
 	 * @access protected
 	 */
 	protected function build( $html ) {
+
+		// Check for data-href property on rendered <div>.
+		if ( preg_match( '/data-href="([^"]+)"/i', $html, $data_href ) ) {
+			$html = $data_href[1];
+		}
+
 		$this->register_json(
 			'json',
 			array(
 				'#url#' => self::_get_facebook_url( strip_tags( $html ) ),
 			)
-	 );
+		);
```

Comment:
This function is a little unclear to me.  It looks like if the data-href property doesn't match, the JSON will still be registered.   Is that the intention?  If not, might just want to make it more clear here, by wrapping the if statement around the register_json call too.

Also if the $html variable is actually just a $url, that could be better named.

Reply:
The `node_matches()` function will match if the FB url is just passed through. Originally the `build()` method was just receiving the FB url, but now it also accepts the rendered oembed markup. In the new `build()` method there is a chance where $html will have the `data-href` attribute and in this case we extract the URL from it. The design pattern for other components uses $html or $text as parameter in the `build()` method. I'd like to keep it as $html in this case, and defer the clarity to the usage of the method `self::_get_facebook_url()`. Does this clarify things @mslinnea?

Reply:
Gotcha, thanks for explaining!